felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1082,6 +1082,20 @@ func (m *bpfEndpointManager) cleanupOldXDPAttach(iface string) error {
 	return nil
 }
 
+func cleanupTcxPins(iface string) {
+	for _, attachHook := range []hook.Hook{hook.Ingress, hook.Egress} {
+		ap := tc.AttachPoint{
+			AttachPoint: bpf.AttachPoint{
+				Iface: iface,
+				Hook:  attachHook,
+			},
+		}
+		if err := ap.DetachTcxProgram(); err != nil {
```

Comment:
in the DetachTcxProgram, what if link.Open or link.Detach fail, you should still call `os.Remove(progPinPath)` right?

Is it worth reporting that the detach failed? The program is quite likely detached, we just need to remove the pin, don't we?